contrib/ci/build_windows.sh

```diff
@@ -57,25 +69,91 @@ echo $VERSION > $DESTDIR/VERSION
 sed -i 's,UpdateMotd=.*,UpdateMotd=false,' $DESTDIR/etc/fwupd/daemon.conf
 
 # create a setup binary
-cd $DESTDIR
-mkdir -p $DESTDIR/setup
-makensis -NOCD $build/contrib/setup-win32.nsi
+CERTDIR=/etc/pki/tls/certs
+MINGW32BINDIR=/usr/x86_64-w64-mingw32/sys-root/mingw/bin
 
-#so that it's actually executable
-cp /usr/x86_64-w64-mingw32/sys-root/mingw/bin/*.dll bin
+# deps
+find $MINGW32BINDIR \
+	-name curl.exe \
+	-o -name gspawn-win64-helper-console.exe \
+	-o -name gspawn-win64-helper.exe \
+	-o -name iconv.dll \
+	-o -name libarchive-13.dll \
+	-o -name libbrotlicommon.dll \
+	-o -name libbrotlidec.dll \
+	-o -name libbz2-1.dll \
+	-o -name libcrypto-1_1-x64.dll \
+	-o -name libcurl-4.dll \
+	-o -name libffi-*.dll \
+	-o -name libgcc_s_seh-1.dll \
+	-o -name libgio-2.0-0.dll \
+	-o -name libglib-2.0-0.dll \
+	-o -name libgmodule-2.0-0.dll \
+	-o -name libgmp-10.dll \
+	-o -name libgnutls-30.dll \
+	-o -name libgnutls-30.dll \
+	-o -name libgobject-2.0-0.dll \
+	-o -name libhogweed-*.dll \
+	-o -name libidn2-0.dll \
+	-o -name libintl-8.dll \
+	-o -name libjson-glib-1.0-0.dll \
+	-o -name liblzma-5.dll \
+	-o -name libnettle-*.dll \
+	-o -name libp11-kit-0.dll \
+	-o -name libpcre-1.dll \
+	-o -name libsqlite3-0.dll \
+	-o -name libssh2-1.dll \
+	-o -name libssl-1_1-x64.dll \
+	-o -name libssp-0.dll \
+	-o -name libtasn1-6.dll \
+	-o -name libusb-1.0.dll \
+	-o -name libwinpthread-1.dll \
+	-o -name libxml2-2.dll \
+	-o -name zlib1.dll \
+	| wixl-heat \
+	-p $MINGW32BINDIR/ \
+	--directory-ref BINDIR \
+	--var "var.MINGW32BINDIR" \
+	--component-group "CG.fwupd-deps" | \
+	tee $build/contrib/fwupd-deps.wxs
 
-#remove static archives
-find -type f -print0 -name "*.dll.a" | xargs rm -f
+# CA bundle
+echo $CERTDIR/ca-bundle.crt | wixl-heat \
+	-p $CERTDIR/ \
+	--directory-ref BINDIR \
+	--var "var.CRTDIR" \
+	--component-group "CG.fwupd-certs" | \
+	tee $build/contrib/fwupd-certs.wxs
 
-#remove stuff that we really don't need
-rm -fr gcab.exe \
-       xb-tool.exe \
-       share/man \
-       include \
-       fwupd \
-       lib/*.a \
-       lib/pkgconfig/ \
-       var
+# our files
+find $DESTDIR | \
+	wixl-heat \
+	-p $DESTDIR/ \
+	-x include/ \
+	-x share/fwupd/device-tests/ \
+	-x share/tests/ \
+	-x share/man/ \
+	-x share/doc/ \
+	-x lib/pkgconfig/ \
+	--directory-ref INSTALLDIR \
+	--var "var.DESTDIR" \
+	--component-group "CG.fwupd-files" | \
+	tee $build/contrib/fwupd-files.wxs
 
-export WINEPATH="/usr/x86_64-w64-mingw32/sys-root/mingw/bin/;$build/libfwupd/;$build/libfwupdplugin/;$build/subprojects/libxmlb/src/;$build/subprojects/gcab/libgcab/;$build/subprojects/libjcat/libjcat/;$build/subprojects/gusb/gusb/"
-ninja -C $build test
+MSI_FILENAME="setup/fwupd-$VERSION-setup-x86_64.msi"
+mkdir -p setup
```

Comment:
I think from this point onwards it would be good to only run the code if `CI=1`.  That way we don't need to make the MSI for "local" builds.

Reply:
I'm not sure much of contrib/ci/build_windows.sh makes sense outside of CI -- it unconditionally `rm -rf`'s your build-win32 for instance. Maybe we make the script quit right at the start if CI != 1?

Reply:
If we want people to be able to make changes to Win32, we have to make building for Win32 hosts accessible though..

Reply:
I don't disagree, but I think that's a different script to the CI one.

Reply:
We do have other scripts that change one or two things based on CI=1 though.

Reply:
I think it actually makes sense to have different compile options for the CI and a local win32 build. I've added a script for local "getting up and going" use.